src/components/AdminPageLayout.js

```diff
@@ -10,12 +13,17 @@ const AdminPageLayout = (props) => {
         {props.children}
       </div>
       <MainFooter />
+      <Spinner
+        canShow={props.spinner.canShow}
+        messageId={props.spinner.messageId}
+      />
     </div>
   );
 };
 
 AdminPageLayout.propTypes = {
+  spinner: PropTypes.object,
   children: PropTypes.element,
 };
 
-export default AdminPageLayout;
+export default connect(mapStateToProps)(AdminPageLayout);
```

Comment:
With these modifications, this is no longer a **stateless component**, because it now depends on the redux store – so it needs to be moved to the `containers` directory.